mlir/lib/Conversion/LLVMCommon/TypeConverter.cpp

```diff
@@ -527,8 +527,11 @@ LLVMTypeConverter::getMemRefAddressSpace(BaseMemRefType type) const {
     return failure();
   if (!(*converted)) // Conversion to default is 0.
     return 0;
-  if (auto explicitSpace = llvm::dyn_cast_if_present<IntegerAttr>(*converted))
-    return explicitSpace.getInt();
+  if (auto explicitSpace = dyn_cast_if_present<IntegerAttr>(*converted)) {
+    if (explicitSpace.getType().isIndex() ||
```

Comment:
Could we instead change this to get the integer out of `uiN` attributes instead?

If not, this is reasonably approved

Reply:
> Thanks, there currently is no separate attributes defined for signless and signed integer in BuiltinAttributes and IntegerAttr represents both signless and signed. [BuiltinAttributes](https://github.com/llvm/llvm-project/blob/4c8779388fd2f25730a1b044f9f10d6717833fce/mlir/include/mlir/IR/BuiltinAttributes.td#L657)

Reply:
Thank you for the reviews. If the PR looks good, can someone help merge because I don't have write access